qcodes/instrument_drivers/tektronix/AWG5014.py

```diff
@@ -341,328 +345,426 @@ def __init__(self, name, setup_folder, address, reset=False,
+        Args:
+            update (bool): whether to return an updated state. Default: True
+
+        Returns:
+
+            dict: a JSON-serialisable dict with all information.
+        """
         return self.snapshot(update=update)
 
     def get_state(self):
-        state = self.ask('AWGC:RSTATE?')
+        """
+        This query returns the run state of the arbitrary waveform
+        generator or the sequencer.
+
+        Returns:
+            str: either 'Idle', 'Waiting for trigger', or 'Running'.
+
+        Raises:
+            ValueError: if none of the three states above apply.
+        """
+        state = self.ask('AWGControl:RSTATe?')
         if state.startswith('0'):
             return 'Idle'
         elif state.startswith('1'):
             return 'Waiting for trigger'
         elif state.startswith('2'):
             return 'Running'
         else:
-            raise ValueError(__name__ + ' : AWG in undefined state "%s"' %
-                             state)
+            raise ValueError(__name__ + (' : AWG in undefined ' +
+                                         'state "{}"').format(state))
 
     def start(self):
-        '''
-        Convenience function, identical to run()
-        '''
+        """Convenience function, identical to self.run()"""
         return self.run()
 
     def run(self):
-        self.write('AWGC:RUN')
+        """
+        This command initiates the output of a waveform or a sequence.
+        This is equivalent to pressing Run/Stop button on the front panel.
+        The instrument can be put in the run state only when output waveforms
+        are assigned to channels.
+
+        Returns:
+            The output of self.get_state()
+        """
+        self.write('AWGControl:RUN')
         return self.get_state()
 
     def stop(self):
-        self.write('AWGC:STOP')
+        """This command stops the output of a waveform or a sequence."""
+        self.write('AWGControl:STOP')
+
+    def force_trigger(self):
+        """
+        This command generates a trigger event. This is equivalent to
+        pressing the Force Trigger button on front panel.
+        """
+        self.write('*TRG')
 
     def get_folder_contents(self, print_contents=True):
+        """
+        This query returns the current contents and state of the mass storage
+        media (on the AWG Windows machine).
+
+        Args:
+            print_contents (bool): If True, the folder name and the query
+                output are printed. Default: True.
+
+        Returns:
+            str: A comma-seperated string of the folder contents.
+        """
         if print_contents:
             print('Current folder:', self.get_current_folder_name())
-            print(self.ask('MMEM:CAT?')
+            print(self.ask('MMEMory:CATalog?')
                   .replace(',"$', '\n$').replace('","', '\n')
                   .replace(',', '\t'))
-        return self.ask('mmem:cat?')
+        return self.ask('MMEMory:CATalog?')
```

Comment:
Yes, that is better. I didn't go through all the methods to see if they were smart, so there are probably (quite?) a few blunders left. I'll certainly fix this one.